strawberryfields/apps/similarity.py

```diff
@@ -412,119 +527,218 @@ def prob_event_mc(
         raise ValueError("Maximum number of photons per mode must be non-negative")
 
     modes = graph.order()
-    A = nx.to_numpy_array(graph)
-    mean_photon_per_mode = n_mean / float(modes)
-
-    p = sf.Program(modes)
-
-    # pylint: disable=expression-not-assigned
-    with p.context as q:
-        sf.ops.GraphEmbed(A, mean_photon_per_mode=mean_photon_per_mode) | q
-
-        if loss:
-            for _q in q:
-                sf.ops.LossChannel(1 - loss) | _q
-
-    eng = sf.LocalEngine(backend="gaussian")
-    result = eng.run(p)
+    State = _get_state(graph, n_mean, loss)
 
     prob = 0
 
-    for _ in range(samples):
+    for _ in range(mc_samples):
         sample = event_to_sample(photon_number, max_count_per_mode, modes)
-        prob += result.state.fock_prob(sample, cutoff=photon_number + 1)
+        prob += State.fock_prob(sample, cutoff=photon_number + 1)
 
-    prob = prob * event_cardinality(photon_number, max_count_per_mode, modes) / samples
+    prob = prob * event_cardinality(photon_number, max_count_per_mode, modes) / mc_samples
 
     return prob
 
 
-def feature_vector_sampling(
-    samples: list, event_photon_numbers: list, max_count_per_mode: int = 2
+def feature_vector_orbits(
+    graph: nx.Graph,
+    list_of_orbits: list,
+    n_mean: float = 5,
+    mc_samples: int = None,
+    loss: float = 0.0,
 ) -> list:
-    r"""Calculates feature vector with respect to input samples.
+    r"""Calculates feature vector made up of either exact or MC-approximated orbit probabilities for the input graph.

```

Comment:
Alternatively, you can omit mentioning exact vs MC here and instead do it in the main part.

Reply:
I will mention it at the start as well. But in case, someone directly goes to the function without reading the module introduction, I think I will keep it in the docstring too.